src/contexts/messageListContext/MessageListContext.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { PropsWithChildren, useContext } from 'react';
+
+import { getDisplayName } from '../utils/getDisplayName';
+
+import type { ChannelState } from 'stream-chat';
+
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../types/types';
+
+export type MessageListContextValue<
```

Comment:
How about we rename this context to `MessagePaginatorContext`

`messagesContext` and `messageListContext` sound very same to me!!

Reply:
Changed to PaginatedMessageListContext to be semantic.